docs/guides/08-interactivity-events.md

```diff
@@ -0,0 +1,411 @@
+    console.log(`Now you can see ${numberOfFeatures} cities`);
+};
+layer.on('updated', displayNumberOfCities);
+```
+> Notice how the variable can be accessed directly from the `carto.Viz` object, inside its `variables` array, without the `@` symbol. Its content is accessible using `.value`, and this is possible cause the expression has no `properties` related to the features themselves.
+
+You can imagine `layer:updated` event as a kind of `layer:viz-updated` event, notifying you whenever something relevant has changed in the viz attached to the layer.
+
+If you want to reduce the number of current `console.log` entries, to better see the new one, you can remove the previous handler on `map:move` with:
+```js
+map.off('move', displayCenter);
+```
+
+You have already advanced a lot in this guide. Now take a small rest and check your work with this:
+<div class="example-map">
+    <iframe
+        id="guides-interactivity-step-3"
+        src="/developers/carto-vl/examples/maps/guides/interactivity/step-3.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### Data-driven variables
+If the data you are interested in for your interaction is a feature `property`, such as the _name_ of the city or its _population_, then you can also use some variables to store them. Those are called _data-driven variables_, cause their values change as you interact with each of the features (in our example, with each city).
+
+To test them you should edit again your viz, so it looks like this:
+```js
+const viz = new carto.Viz(`
+    @currentFeatures: viewportFeatures()
+    @name: $name
+    @popK: $pop_max / 1000.0
+`);
+```
+> Both properties, $name and $pop_max are columns in the original dataset.
+
+As the variables depend on properties, you can't just access them using something like `viz.variables.name.value` on `layer:updated`. That will throw an error saying: _property needs to be evaluated in a 'feature'_. You need to use [carto.Interactivity](/developers/carto-vl/reference/#cartointeractivity).
+
+
+### Feature events
+All feature interactions are ruled by the `carto.Interactivity`, so let's create an object of this type, associating it with the current layer.
+
+Add this line:
+```js
+const interactivity = new carto.Interactivity(layer);
+```
+
+Then use its `featureClick` event to react when you click on a city:
+```js
+interactivity.on('featureClick', featureEvent => {
+    featureEvent.features.forEach((feature) => {
+        const name = feature.variables.name.value;
+        const popK = feature.variables.popK.value.toFixed(0);
+        console.log(`You have clicked on ${name} with a population of ${popK}K`);
+    });
+});
+```
+> Notice how `carto.Interactivity` provides you for free with a dynamic change on the mouse pointer when you hover on a feature, and how it handles a collection, cause you can click on several features at the same time if they are near enough.
```

Comment:
provides you for free with a dynamic change -> provides you with a dynamic change